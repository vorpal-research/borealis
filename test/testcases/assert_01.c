
#include <stdlib.h>

int main(int argc, char* argv[]) {
	int* val = (int*)malloc(sizeof(int));
	
	// @assert val != 0

	*val = 2 + argv[0][0];

	return *val+1;
}