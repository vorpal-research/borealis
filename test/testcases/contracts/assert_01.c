#include <stdlib.h>

int main(int argc, char* argv[]) {
	// @assume argv != 0
	// @assume *argv != 0

	int* val = (int*)malloc(sizeof(int));
	// @assert val != 0

	// @assume val != 0
	*val = 2 + argv[0][0];

	return *val+1;
}
