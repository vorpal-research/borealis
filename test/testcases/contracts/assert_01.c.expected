
[
	
	{
		"location" : 
		{
			"filename" : "test/testcases/contracts/assert_01.c",
			"loc" : 
			{
				"col" : 2,
				"line" : 8
			}
		},
		"type" : "ASR-01"
	}
]
